### ISSUE TITLE
mixins-before-declarations Errors

### ISSUE THREAD

**Issue by kamalbennani** [sasstools/sass-lint#230]
Hello,
When i used you sass lint, i found this error which i found little bit confused, normally it's shouldn't cause this error.
Could you please explain to me the problem ?
Thanks

```SASS
.area-current {
  $active: #00acac;
  $background: #2d353c;

  @include area-menu($active, $background);
}

.area-new {
  $active: #348fe2;
  $background: #2e475d;

  @include area-menu($active, $background);
}

.area-disappeared {
  $active: #727cb6;
  $background: #424a61;

  @include area-menu($active, $background);
}
```

```TXT
botify-report/sass/_themes.scss
   96:3  error  Mixins should come before declarations  mixins-before-declarations
  103:3  error  Mixins should come before declarations  mixins-before-declarations
  110:3  error  Mixins should come before declarations  mixins-before-declarations
```

**Comment by DanPurdy** [comment]
I'm gonna label this is as a bug but I think it needs further discussion too. Is there any reason you are declaring variabled in each block like that? There's not really any benefit to using them in that code. 

Technically a variable declaration is classed as a declaration and so should come after a mixin, now obviously in the case you have this wouldn't be practical. Is there a lot more going on that Is using those variables within that block?

**Comment by bgriffith** [comment]
No. Should only cause an error if it's a propert being set in my opinion.

**Comment by BPScott** [comment]
> I'm gonna label this is as a bug but I think it needs further discussion too. Is there any reason you are declaring variables in each block like that? There's not really any benefit to using them in that code.

Variables defined within a block are useful as they are scoped to that block.

I agree with @bgriffith: this should only be an error when a mixin appears before a property or a nested selector by default.
